stdlib/REPL/src/REPLCompletions.jl

```diff
@@ -213,8 +214,26 @@ function complete_symbol(sym::String, @nospecialize(ffunc), context_module::Modu
                 end
             end
         end
+
+        # Also, try abstract-interpreting propertynames
+        thunk = context_module.eval(:(() -> propertynames($(ex))))
+        code_info, rett = Core.Compiler.typeinf_code(
+            Core.Compiler.NativeInterpreter(),
+            first(methods(thunk)),
+            Tuple{typeof(thunk)},
+            Core.svec(),
+            true,
+        )
+        if rett <: Tuple && hasproperty(rett, :parameters) && all(rett.parameters .<: Symbol)
+            for field in code_info.code[end].val
```

Comment:
This assumes that the code constant folds, which is not a valid assumption. You need to get the non-wiedened `Const` lattice element out of inference and look at that instead.